### PR TITLE
Frontend: Enable `$GlobalActors` by default

### PR DESCRIPTION
**Pull request by tshortli**
Global actors have been an accepted language feature for a couple of compiler releases now, but the feature definition was associated with the `--enable-experimental-concurrency` flag. This caused some `.swiftinterface`s containing global actor declarations to be unparsable because the logic for surrounding declarations with the `$GlobalActors` feature guard was incomplete (for example, classes with a global actor attribute were not guarded even though the declarations of global actor types were).
    
Rather than trying to fix the logic of `usesFeatureGlobalActors()`, enable it by default.
    
Adds a test that demonstrates that modules defining and using public global actors produce module interfaces that can be parsed successfully.
    
Resolves rdar://100150703

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please test

File: lib/IRGen/GenStruct.cpp [hunk 1]
```diff
@@ -326,10 +326,9 @@ namespace {
   };
 
   /// A type implementation for loadable record types imported from Clang.
-  class LoadableClangRecordTypeInfo final :
-    public StructTypeInfoBase<LoadableClangRecordTypeInfo, LoadableTypeInfo,
-                              ClangFieldInfo> {
-    IRGenModule &IGM;
```
Comment (tshortli) [inline lib/IRGen/GenStruct.cpp:332]: Drive-by fix to silence a warning about an unused member.

**Comment by tshortli** [comment]
@swift-ci please test

**Comment by tshortli** [comment]
I suppose we should also enable `$Actors` by default but I think that could be a separate PR.

**Review by DougGregor — APPROVED** [review]

**Comment by tshortli** [comment]
@swift-ci please test

**Comment by tshortli** [comment]
This fixed https://github.com/apple/swift/issues/62370
